src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -892,6 +895,42 @@ private String getNearestClassFrameName() {
         return frame.getFrameName();
     }
 
+    /**
+     * Checks if the token is a Lambda parameter.
+     * @param ast the {@code DetailAST} value of the token to be checked
+     * @return true if the token is a Lambda parameter
+     */
+    private static boolean isLambdaParameter(DetailAST ast) {
+        DetailAST parent;
+        for (parent = ast.getParent(); parent != null;
+             parent = parent.getParent()) {
+            if (parent.getType() == TokenTypes.LAMBDA) {
+                break;
+            }
+        }
+        final boolean returnValue;
+        if (parent == null) {
+            returnValue = false;
+        }
+        else if (ast.getType() == TokenTypes.PARAMETER_DEF) {
+            returnValue = true;
+        }
+        else {
+            final DetailAST lambdaParameters = parent.findFirstToken(TokenTypes.PARAMETERS);
+            if (lambdaParameters == null) {
+                returnValue = parent.getFirstChild().getText().equals(ast.getText());
```

Comment:
@MEZk 
Here. This is how I'd coded it initially. The intention was to compare the variable names rather than objects/references (ast nodes). So using `==` wouldn't work. Correct me if I am wrong.